File: app/src/main/java/com/example/david/lists/view/addedit/common/IAddEditContract.java
package com.example.david.lists.view.addedit.common;

public interface IAddEditContract {
    interface View {
        void setStateError(String message);

        void finishView();
    }

    interface Logic {
        String getCurrentTitle();

        void validateInput(String input);
    }

    interface ViewModel {
        void setTaskType(TaskType taskType);

        TaskType getTaskType();

        void setId(String id);

        void setCurrentTitle(String title);

        void setUserListId(String userListId);

        void setPosition(int position);

        String getId();

        String getCurrentTitle();

        String getUserListId();

        int getPosition();

        String getMsgEmptyTitle();

        String getMsgTitleUnchanged();
    }

    enum TaskType {
        TASK_ADD,
        TASK_EDIT
    }
}

File: app/src/main/java/com/example/david/lists/view/addedit/common/IAddEditContract.kt
package com.example.david.lists.view.addedit.common

interface IAddEditContract {
    interface View {
        fun setStateError(message: String)

        fun finishView()
    }

    interface Logic {
        val currentTitle: String

        fun validateInput(input: String)
    }

    interface ViewModel {
        var taskType: TaskType

        var id: String

        var currentTitle: String

        var userListId: String?

        var position: Int

        val msgEmptyTitle: String

        val msgTitleUnchanged: String
    }

    enum class TaskType {
        ADD,
        EDIT
    }
}
